include/envoy/protobuf/message_validator.h

```diff
@@ -33,8 +33,16 @@ class ValidationVisitor {
    * @param description human readable description of the field
    */
   virtual void onUnknownField(absl::string_view description) PURE;
+
+  /**
+   * If true, skip this validation visitor in the interest of speed when
+   * possible.
+   **/
+  virtual bool skipValidation() { return false; }
```

Comment:
We generally prefer pure virtual interfaces to remain pure, and push this default to the implementations.